ffi/src/lib.rs

```diff
@@ -293,9 +293,10 @@ pub extern "C" fn wirefilter_add_type_field_to_scheme(
     name_ptr: *const c_char,
     name_len: usize,
     ty: CType,
+    optional: bool,
```

Comment:
nit: Looking at this now, I think I actually slightly prefer a separate method for adding an optional field. Because otherwise, it's not super obvious what the `true`/`false` is doing when you look at calls.

I guess the other (small) advantage is that it would be less of a breaking change, as the existing method could keep the same signature.